src/it/scala/com/exasol/spark/PredicatePushdownSuite.scala

```diff
@@ -51,4 +48,38 @@ class PredicatePushdownSuite extends FunSuite with BaseDockerSuite with DataFram
     assert(result === Set((1, "Berlin"), (2, "Paris")))
   }
 
+  test("date and timestamp should be read and filtered correctly") {
+    import java.sql.Date
+
+    createDummyTable()
+    val df = spark.read
+      .format("exasol")
+      .option("host", container.host)
+      .option("port", s"${container.port}")
+      .option("query", s"SELECT date_info, updated_at FROM $EXA_SCHEMA.$EXA_TABLE")
+      .load()
+
+    val resultDate = df.collect().map(_.getDate(0))
+    assert(resultDate.contains(Date.valueOf("2017-12-31")))
+
+    val minTimestamp = Date.valueOf("2017-12-30")
+    val resultTimestamp = df.collect().map(_.getTimestamp(1)).map(x => x.after(minTimestamp))
+    assert(!resultTimestamp.contains(false))
+
+    val filteredDf = df
+      .filter(col("date_info") === "2017-12-31")
+      .filter(col("updated_at") > "2017-12-31 00:00:00.000")
+    assert(filteredDf.count() === 1)
```

Comment:
Somehow Spark does not create a filter if types are not equal. 

Could you please change above conditions as below? For example:
```scala
val filteredDf = df                                                                               
    .filter(col("date_info") === Date.valueOf("2017-12-31"))                                     
    .filter(col("updated_at") > Timestamp.valueOf("2017-12-31 00:00:00.000")) 
```

You can check the final generated (or enriched) query string from [this line](https://github.com/EXASOL/spark-exasol-connector/blob/master/src/main/scala/com/exasol/spark/ExasolRelation.scala#L60)

Could you please also separate those two check? For example,

```scala
val filteredByDateDF = df                                                                               
    .filter(col("date_info") === Date.valueOf("2017-12-31"))                     
assert(filteredByDateDF.count() === 1)   
```
and 
```scala
val filteredByTimestampDF = df                                                                                                                 
    .filter(col("updated_at") < Timestamp.valueOf("2017-12-31 00:00:00.000"))                       
assert(filteredByTimestampDF.count() === 0)   
```